internal/controller/etcdcluster_controller.go

```diff
@@ -387,9 +418,28 @@ func (r *EtcdClusterReconciler) getClusterStateConfigMapName(cluster *etcdaenixi
 	return cluster.Name + "-cluster-state"
 }
 
+// updateClusterState patches status condition in cluster using merge by Type
+func (r *EtcdClusterReconciler) updateClusterState(cluster *etcdaenixiov1alpha1.EtcdCluster, state metav1.Condition) {
+	if initIdx := slices.IndexFunc(cluster.Status.Conditions, func(condition metav1.Condition) bool {
+		return condition.Type == state.Type
+	}); initIdx != -1 {
+		cluster.Status.Conditions[initIdx].Status = state.Status
+		cluster.Status.Conditions[initIdx].LastTransitionTime = state.LastTransitionTime
+		cluster.Status.Conditions[initIdx].ObservedGeneration = cluster.Generation
+		cluster.Status.Conditions[initIdx].Reason = state.Reason
+		cluster.Status.Conditions[initIdx].Message = state.Message
+	} else {
+		state.ObservedGeneration = cluster.Generation
+		cluster.Status.Conditions = append(cluster.Status.Conditions, state)
+	}
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *EtcdClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&etcdaenixiov1alpha1.EtcdCluster{}).
```

Comment:
For(&etcdaenixiov1alpha1.EtcdCluster{},
builder.WithPredicates(predicate.GenerationChangedPredicate{})).